ansible/roles/test/files/ptftests/advanced-reboot.py

```diff
@@ -989,6 +989,8 @@ def runTest(self):
                 self.handle_warm_reboot_health_check()
             self.handle_post_reboot_health_check()
 
+            # Check sonic version after reboot
+            self.check_sonic_version_after_reboot()
```

Comment:
Minor concern in moving `check_sonic_version_after_reboot` towards the end is that even if upgrade fails now the health checks will be performed.
Earlier if the upgrade failure was seen `thread.interrupt_main()` was called and it would crash the main thread, skipping all the checks which are unnecessary if the upgrade itself has failed.

May be adding right after `self.wait_until_reboot()` and before health_checks will be better?

Reply:
It's a good suggestion. But I don't think checking sonic version right after ```wait_until_reboot``` is reliable.  The ```wait_until_reboot``` only confirms the DUT is down, but we are not sure if it's up. So I place the check just at the very end of the test.